### ISSUE TITLE
hcaptcha error after solving the captcha

### ISSUE THREAD

**Issue by loquesangra** [taylordotfish/harmony#14]
Successfully solved the CAPTCHA challenge.
[proxychains] Strict chain  ...  127.0.0.1:4321  ...  discordapp.com:443  ...  OK
```
Login failed. Errors: {
  "captcha_key": [
    "invalid-input-response"
  ],
  "captcha_sitekey": "f5561ba9-8f1e-40ca-9b5b-a0b3f719ef34",
  "captcha_service": "hcaptcha"
}
```

**Comment by taylordotfish** [comment]
Yeah, it looks like Discord now uses hCaptcha instead of reCAPTCHA. A non-ideal workaround is to first log in or register through the official web client and complete the CAPTCHA there, which will then likely let Harmony log in without having to complete another CAPTCHA. But of course, this requires running proprietary JavaScript.

I'd like to look into adding support for hCaptcha, but I don't have much time for that right now (but, as always, I'd gladly welcome any PRs).

**Comment by lua-rocks** [comment]
Today I was in a good mood and decided to give this proprietary spyware a chance, but well, if Discord itself is not happy to me, then I have to return its ip-addresses to /etc/hosts forever.

**Comment by JacobKfromIRC** [comment]
Has anyone made any progress with this? I'm trying to figure out how to make hCaptcha work with my own JavaScript and I'm stuck trying to figure out how the "n" parameter is generated, which is needed to actually get the puzzle images. Somehow hsj.js takes the "req" JWT token and turns it into "n", using some randomness so it's different every time. I think it's some sort of proof of work system, possibly Hashcash, based on info from `git clone https://github.com/Implex-ltd/hcaptcha-reverse` (not licensed), though that document focuses more on hsw.js than hsj.js so it might not be the same for both. I think hsw.js is for browsers supporting WebAssembly and hsj.js is for browsers not supporting WebAssembly, but that's just a guess.
